samples: matter: added support for nRF54LC10 (#30038)
nrfconnect/sdk-nrf: doc/nrf/includes/matter/interface/nfc.txt

NFC port with antenna attached:
    Optionally used for obtaining the onboarding information from the Matter accessory device to start the commissioning the device procedure while using a commercial ecosystem.
    See the `Testing with commercial ecosystem`_ section.

    .. note::

>        The nRF54LC10A SoC doesn't have a NFC peripheral, therefore this feature is not supported on nRF54LC10 DK.

Reviewer: ```suggestion
       The nRF54LC10A SoC does not have an NFC peripheral, thus this feature is not supported on the nRF54LC10 DK.
```
